### DISCUSSION TITLE
Navigation rerouting fast

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#361] [Q&A] [answered]
Generating a Brouter design to a single long-range target takes a long time.
(Design without additional intermediate manually placed shaping points)

Proposal: Send a virtual reference rider target xx km ahead.
Autorecalculate to this target except if there is a VIA STRICT in between.
Just the first xx km are so to be recalculated the rest remains unchanged.

**Comment by devemux86** [comment]
@0709wiwiwi thanks for the interesting idea.

Essentially you suggest that the rerouting could be done until a distance ahead (e.g. 10 km).
(unless there is a via / shaping point between)

So the rerouting will be extremely fast as only this short distance will be recalculated.

**Reply by 0709wiwiwi** [reply]
I would argue except only to a STRICT VIA. This prevents the autorecalculation target is toward an already very nearby (non strict) shaping point.

**Reply by devemux86** [reply]
@0709wiwiwi I changed the title to better describe the idea.

**Reply by Leeb64** [reply]
Very good idea.
Instead of 10km away, the next intersection or the one after that wouldn't be bad either....
or the next intersection after 10km...
This would prevent you from being “directed back” to the “virtual driver” after an intersection.

**Reply by 0709wiwiwi** [reply]
> So the rerouting will be extremely fast as only this short distance will be recalculated.

**Reply by Leeb64** [reply]
![beispiel](https://github.com/devemux86/cruiser/assets/120722053/18bb757e-4c18-4d3a-a5f5-55a285eec9ae)

Example:
You want to go from Sentheim to the Ballon d Alsace.
There is a closure at Masevaux.(Distance Masevaux-Ballon ca.20km)
The “Virtual Driver” is now in Dolleren.
Cruiser now redirects to Dolleren...
To do this, the route between Dolleren and the Ballon must then be driven twice.
Therefore the next intersection after 10km.
Cases like this happen often...
The simplest option would be to increase the distance to the "virtual driver", perhaps to 30-50km. That would reduce such situations and the calculation time would still be short.

**Comment by devemux86** [comment]
We could also use different options for the rerouting, e.g. do not use avoid options for fastest result.

Something like that was also discussed here:
- https://github.com/devemux86/cruiser/discussions/271

**Reply by Leeb64** [reply]
The rerouter profile takes about 1/3 of the calculation time.
For 100km approx.3sec.
A combination of rerouter profile and “virtual driver” with a distance of 50km(ca 1s) would be a solution.
But the rerouter profile uses EVERY path mercilessly - whether allowed or not!
This is suboptimal.
The normal driving cruiser profile without avoidances is just as fast and uses legal roads.
A combination of Cruiser-driver profile(without avoidances) and “virtual driver” with a distance of 50km would be the better solution.

**Reply by devemux86** [reply]
It is like the [shortest profile](https://github.com/abrensch/brouter/blob/master/misc/profiles2/shortest.brf), mostly useful on pedestrian travel modes.

> The normal driving cruiser profile without avoidances is just as fast and uses legal roads.

Indeed it should be.

> A combination of Cruiser-driver profile(without avoidances) and “virtual driver” with a distance of 50km would be the better solution.

Probably this could be the optimal solution.

**Reply by Leeb64** [reply]
If the virtual driver's 50km lead is not enough (la Bonette), an option could be to display the detour route in a different color (purple or something like that).
You would quickly recognize an overlap.

**Comment by devemux86** [comment]
Disabling temporarily the avoid options in **rerouting** until next waypoint is easy, so the partial route calculation is faster.

Should we disable all of them or keep some major avoid options?
e.g. avoid motorways, toll roads, ferries

**Reply by DolomitesTom** [reply]
I won't like this disabling options for rerouting. This is quick enough for me. I don't like unwanted effects.

**Comment by devemux86** [comment]
I did put a fast text note together see in attachment.  Please use translate.  
I put to much time in multiple gps forums, have also other things to do sometimes.

[Intelligent to recalculate.txt](https://github.com/devemux86/cruiser/files/14421854/Intelligent.to.recalculate.txt)

_Originally posted by @0709wiwiwi in https://github.com/devemux86/cruiser/discussions/142#discussioncomment-8606742._

**Reply by devemux86** [reply]
Some notes:

- This is not related to the route storage format, it is just about the route calculation.

- This is not related to the new [intelligent route calculation](https://github.com/devemux86/cruiser/discussions/142).

- It is about the **rerouting in navigation** which already works in intelligent mode all these years.
Navigation calculates a rerouting until next waypoint and the _route remains unchanged_.

- Essentially, it is about rerouting to a new virtual point a fixed distance ahead.

**Comment by devemux86** [comment, marked as answer]
This should be covered by:
- https://github.com/devemux86/cruiser/discussions/499
